contracts/contracts/PoolManagerLite.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "./Interfaces.sol";
+import "@openzeppelin/contracts-0.6/math/SafeMath.sol";
+import "@openzeppelin/contracts-0.6/token/ERC20/IERC20.sol";
+
+/**
+ * @title   PoolManagerLite
+ * @author  ConvexFinance
+ * @notice  Pool Manager Lite
+ * @dev     Add pools to the Booster contract
+ */
+contract PoolManagerLite {
+    using SafeMath for uint256;
+
+    address public immutable booster;
+    address public operator;
+    bool public protectAddPool;
+    bool public isShutdown;
+
+    constructor(address _booster) public {
+        booster = _booster;
+        protectAddPool = true;
+        operator = msg.sender;
+    }
+
+    function setOperator(address _operator) external {
+        require(msg.sender == operator, "!auth");
+        operator = _operator;
+    }
+
+    function setProtectPool(bool _protectAddPool) external {
+        require(msg.sender == operator, "!auth");
+        protectAddPool = _protectAddPool;
+    }
+
+    function addPool(address _gauge) external returns (bool) {
+        return _addPool(_gauge, 3);
+    }
+
+    function addPool(address _gauge, uint256 _stashVersion) external returns (bool) {
+        return _addPool(_gauge, _stashVersion);
+    }
+
+    function _addPool(address _gauge, uint256 _stashVersion) internal returns (bool) {
+        require(!IPools(booster).gaugeMap(_gauge), "already registered gauge");
+
+        if (protectAddPool) {
+            require(msg.sender == operator, "!auth");
+        }
+
+        address lptoken = ICurveGauge(_gauge).lp_token();
+        require(!IPools(booster).gaugeMap(lptoken), "already registered lptoken");
```

Comment:
The gauge map only tracks the gauges address not the lp token , right ? 
```
        require(!IPools(booster).gaugeMap(lptoken), "already registered lptoken");
```

Reply:
//must also check that the lp token is not a registered gauge
        //because curve gauges are tokenized